android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt

```diff
@@ -130,6 +131,7 @@ class AudioRecorder : PluginRegistry.RequestPermissionsResultListener {
         grantResults: IntArray
     ): Boolean {
         return if (requestCode == RECORD_AUDIO_REQUEST_CODE) {
+            successCallback?.onSuccess(grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED)
```

Comment:
No @ujas-m-simformsolutions  it will call no matter either granted the permission or decline the permission. it will return the result based on user action.